custom_components/unifiprotect/config_flow.py

```diff
@@ -36,79 +38,126 @@ class UnifiProtectFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+                CONF_DOORBELL_TEXT: "",
+            },
+        )
 
+    @callback
+    async def _async_get_nvr_data(
+        self,
+        user_input: Dict[str, Any],
+    ) -> Tuple[Optional[Dict[str, Any]], Dict[str, str]]:
         session = async_create_clientsession(
             self.hass, cookie_jar=CookieJar(unsafe=True)
         )
 
-        unifiprotect = UpvServer(
-            session,
-            user_input[CONF_HOST],
-            user_input[CONF_PORT],
-            user_input[CONF_USERNAME],
-            user_input[CONF_PASSWORD],
+        if CONF_HOST in user_input:
+            host = user_input[CONF_HOST]
+            port = user_input.get(CONF_PORT, DEFAULT_PORT)
+        else:
+            return None, {}
+
+        protect = UpvServer(
+            session=session,
+            host=host,
+            port=port,
+            username=user_input[CONF_USERNAME],
+            password=user_input[CONF_PASSWORD],
         )
 
+        errors = {}
+        nvr_data = None
         try:
-            server_info = await unifiprotect.server_information()
-            if server_info["server_version"] < MIN_REQUIRED_PROTECT_V:
-                _LOGGER.debug("UniFi Protect Version not supported")
-                errors["base"] = "protect_version"
-                return await self._show_setup_form(errors)
-
+            nvr_data = await protect.server_information()
         except NotAuthorized as ex:
             _LOGGER.debug(ex)
-            errors["base"] = "connection_error"
-            return await self._show_setup_form(errors)
+            errors[CONF_PASSWORD] = "invalid_auth"
         except NvrError as ex:
             _LOGGER.debug(ex)
             errors["base"] = "nvr_error"
-            return await self._show_setup_form(errors)
+        else:
+            if nvr_data["server_version"] < MIN_REQUIRED_PROTECT_V:
+                _LOGGER.debug("UniFi Protect Version not supported")
+                errors["base"] = "protect_version"
 
-        unique_id = server_info[SERVER_ID]
-        server_name = server_info[SERVER_NAME]
+        return nvr_data, errors
 
-        await self.async_set_unique_id(unique_id)
-        self._abort_if_unique_id_configured()
+    async def async_step_reauth(self, user_input: Dict[str, Any] = None) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
 
-        return self.async_create_entry(
-            title=server_name,
-            data={
-                CONF_ID: server_name,
-                CONF_HOST: user_input[CONF_HOST],
-                CONF_PORT: user_input[CONF_PORT],
-                CONF_USERNAME: user_input.get(CONF_USERNAME),
-                CONF_PASSWORD: user_input.get(CONF_PASSWORD),
-            },
-            options={
-                CONF_DISABLE_RTSP: False,
-                CONF_DOORBELL_TEXT: "",
-            },
+        errors = {}
+
+        entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
```

Comment:
I think we still need the https://github.com/home-assistant/core/blob/dev/homeassistant/components/ambee/config_flow.py#L74 type logic since I don't think the `self.context` will still have `entity_id` in there if they have to retry because they entered the password wrong